### PR TITLE
Logrotate

### PR DESCRIPTION
**Pull request by ewoutp**
Added support for rotating log files of started components.

- On `SIGHUP` 
- On interval specified by `--log.rotate-interval`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# Changes from version 0.10.3 to 0.10.4
+# Changes from version 0.10.3 to master
+
+- Added support for log file rotation for started server components.
+
+# Changes from version 0.10.4 to 0.10.4
```
Comment (neunhoef) [inline CHANGELOG.md]: Looks like it should be "0.10.3 to 0.10.4"

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-# Changes from version 0.10.3 to 0.10.4
+# Changes from version 0.10.3 to master
```
Comment (neunhoef) [inline CHANGELOG.md]: Looks like it should read "0.10.4 to master"

File: README.md [hunk 3]
```diff
@@ -490,6 +490,17 @@ debugging only.
 
 show more information (default false).
 
+* `--log.rotate-files-to-keep=int`
+
+set the number of old log files to keep when rotating log files of server components.
```
Comment (neunhoef) [inline README.md]: Maybe mention the default here.

Comment (ewoutp) [reply README.md]: done

File: CHANGELOG.md (hunk 1 above)
Comment (ewoutp) [reply CHANGELOG.md]: both done

**Pull request merged**